clients/include/testing_symv.hpp

```diff
@@ -76,91 +142,122 @@ void testing_symv(const Arguments& arg)
     // Initial Data on CPU
     rocblas_seedrand();
     rocblas_init_symmetric<T>(hA, N, lda);
-    rocblas_init<T>(hx, 1, N, incx);
-    rocblas_init<T>(hy, 1, N, incy);
+    rocblas_init<T>(hx, 1, N, abs_incx);
+    rocblas_init<T>(hy, 1, N, abs_incy);
 
-    // copy vector is easy in STL; hz = hy: save a copy in hz which will be output of CPU BLAS
-    hz = hy;
+    // make copy in hg which will later be used with CPU BLAS
+    hg  = hy;
+    hy2 = hy; // device memory re-test
 
-    // copy data from CPU to device
-    hipMemcpy(dA, hA, sizeof(T) * lda * N, hipMemcpyHostToDevice);
-    hipMemcpy(dx, hx, sizeof(T) * N * incx, hipMemcpyHostToDevice);
-    hipMemcpy(dy, hy, sizeof(T) * N * incy, hipMemcpyHostToDevice);
-
-    /* =====================================================================
-           ROCBLAS
-    =================================================================== */
-    if(arg.timing)
+    if(arg.unit_check || arg.norm_check)
     {
-        gpu_time_used = get_time_us(); // in microseconds
-    }
+        cpu_time_used = get_time_us();
 
-    for(int iter = 0; iter < 1; iter++)
-    {
-        CHECK_ROCBLAS_ERROR(
-            rocblas_symv<T>(handle, uplo, N, &alpha, dA, lda, dx, incx, &beta, dy, incy));
-    }
-    if(arg.timing)
-    {
-        gpu_time_used  = get_time_us() - gpu_time_used;
-        rocblas_gflops = symv_gflop_count<T>(N) / gpu_time_used * 1e6 * 1;
+        // cpu reference
+        cblas_symv<T>(uplo, N, alpha[0], hA, lda, hx, incx, beta[0], hg, incy);
+
+        cpu_time_used = get_time_us() - cpu_time_used;
+        cblas_gflops  = symv_gflop_count<T>(N) / cpu_time_used * 1e6;
     }
 
-    // copy output from device to CPU
-    hipMemcpy(hy, dy, sizeof(T) * N * incy, hipMemcpyDeviceToHost);
+    // clear non-fill half
+    rocblas_clear_symmetric(uplo, hA.data(), N, lda);
```

Comment:
Ah, I see! Initializing it as symmetric makes sense to me now haha. Still don't think we need to use the symmetric initialization in sbmv or spmv, though.

Also, should this be done before the call to `cblas_symv` so we're actually passing in the same information to the cblas version and our version?